File: ios/Sources/App/ViewModels/AgentPresetStore.swift
import Foundation

@MainActor
final class AgentPresetStore: ObservableObject {
    @Published private(set) var presets: [AgentPreset] = []
    private let key = "dev.termvault.agentPresets.v1"

    init() {
        if let data = UserDefaults.standard.data(forKey: key),
           let values = try? JSONDecoder().decode([AgentPreset].self, from: data) {
            presets = values
        } else {
            presets = [
                AgentPreset(name: "Codex", tool: .codex),
                AgentPreset(name: "Claude", tool: .claude),
                AgentPreset(name: "Hermes", tool: .hermes),
                AgentPreset(name: "Project Shell", tool: .shell),
            ]
        }
    }

    func save(_ preset: AgentPreset) {
        if let index = presets.firstIndex(where: { $0.id == preset.id }) { presets[index] = preset }
        else { presets.append(preset) }
        persist()
    }

    func delete(at offsets: IndexSet) {
        for index in offsets.sorted(by: >) { presets.remove(at: index) }
        persist()
    }

    private func persist() {
        if let data = try? JSONEncoder().encode(presets) { UserDefaults.standard.set(data, forKey: key) }
    }
}

File: android/app/src/main/kotlin/dev/termvault/app/workspace/AgentPresetStore.kt
package dev.termvault.app.workspace

import dev.termvault.app.data.db.AgentPresetDao
import dev.termvault.app.data.db.AgentPresetEntity
import dev.termvault.app.data.model.AgentTool
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.flow.SharingStarted
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.flow.stateIn
import kotlinx.coroutines.launch

/** Mirrors iOS `AgentPresetStore`, backed by Room; seeds one default preset per built-in tool on first launch. */
class AgentPresetStore(private val dao: AgentPresetDao, scope: CoroutineScope) {
    val presets: StateFlow<List<AgentPresetEntity>> =
        dao.observeAll().stateIn(scope, SharingStarted.Eagerly, emptyList())

    init {
        scope.launch(Dispatchers.IO) {
            if (dao.count() == 0) {
                dao.upsert(AgentPresetEntity(name = "Codex", toolRaw = AgentTool.CODEX.rawValue))
                dao.upsert(AgentPresetEntity(name = "Claude", toolRaw = AgentTool.CLAUDE.rawValue))
                dao.upsert(AgentPresetEntity(name = "Hermes", toolRaw = AgentTool.HERMES.rawValue))
                dao.upsert(AgentPresetEntity(name = "Project Shell", toolRaw = AgentTool.SHELL.rawValue))
            }
        }
    }

    suspend fun save(preset: AgentPresetEntity) {
        dao.upsert(preset)
    }

    suspend fun delete(preset: AgentPresetEntity) {
        dao.delete(preset)
    }
}
